force-app/main/default/classes/AFFL_MultiRecordTypeMapper.cls

```diff
@@ -330,32 +313,30 @@ public virtual with sharing class AFFL_MultiRecordTypeMapper {
 
     /*******************************************************************************************************
      * @description Whether or not the affiliation requires a program enrollment to be auto-created.
-     * @param affl The affiliation to check for program enrollment creations
+     * @param affiliation The affiliation to check for program enrollment creations
      * @return Whether or not the affiliation requires a program enrollment to be auto-created.
      ********************************************************************************************************/
-    public virtual Boolean needsProgramEnrollment(Affiliation__c affl) {
-        if (affl.Affiliation_Type__c == null) {
+    public virtual Boolean needsProgramEnrollment(Affiliation__c affiliation) {
+        if (affiliation?.Affiliation_Type__c == null) {
```

Comment:
Note that we're checking to see if affiliation is null and if Affiliation_Type__c is null in one go here.